api/src/modules/h3-data/h3-data.service.ts

```diff
@@ -352,4 +408,22 @@ export class H3DataService {
       },
     };
   }
+
+  async getAvailableYearsForH3MaterialData(
+    materialId: string,
+    materialType: MATERIAL_TO_H3_TYPE,
+  ): Promise<number[]> {
+    return await this.h3DataRepository.getAvailableYearsForH3MaterialData(
+      materialId,
+      materialType,
+    );
+  }
+
+  async getAvailableYearsForH3IndicatorData(
+    indicatorId: string,
+  ): Promise<number[]> {
+    return await this.h3DataRepository.getAvailableYearsForH3IndicatorData(
+      indicatorId,
+    );
+  }
```

Comment:
Maybe both of these methods would exist on the 

`h3-filter-years-by-layer.service.ts` ? (maybe with better naming :))

As these are also related getting available years

Reply:
Yes, at first I used this service, but then due to failing test and code investigation I found out that I cannot use it - because: 
- h3-filter-years-by-layer is retrieving available years for the layer (for example, material)
- h3DataIds is an optional parameter of this service
- the service tries to find h3DataIds by the given material Id, if no h3DatIds are found - the server just doesn't include the filter by h3DataIds in the query and returns all the years available for the layer (material)  in general. In our case it is not correct, because we need available years for the particular material and its H3Data, if no h3Data is found, no years shall be returned and the error should be thrown (saying there is no h3 data for that material for that year or any other year in the past)

Reply:
Sorry. maybe my comment was not clear (probably)

What I meant is we could encapsulate your logics:
`getAvailableYearsForH3IndicatorData`
and
`getAvailableYearsForH3MaterialData`

into the standalone service which does stuff about years, so we can have all stuff related to years encapsulated in the same class (the class itself does not receive any params related to years, layer or anything, just its dependencies by injection)

However this is code stylish stuff, let's see the team's thoughts

Reply:
implementation detail, either way is fine, leave it as it is